Unify dense and sparse tests

levsnv opened the pull request:
test cases are almost entirely disjoint, no point creating mechanisms to run a case in both or one sparsity only

levsnv committed:
unified dense adn sparse tests; test cases are almost entirely disjoint

levsnv committed:
stray comment

canonizer commented on cpp/test/sg/fil_test.cu:
```diff
@@ -714,22 +691,30 @@ class BasePredictSparseFilTest : public BaseFilTest {
+    if constexpr (!node_traits<fil_node_t>::IS_DENSE) {
+      dense2sparse();
+    } else {
+      sparse_nodes = nodes;
```
`sparse_node` is no longer a good name, since the nodes can as well be dense.

canonizer reviewed (approved):
Approved, provided that the comments are addressed.

pentschev committed:
Update to UCX-Py 0.24 (#4396)



Authors:
  - Peter Andreas Entschev (https://github.com/pentschev)

Approvers:
  - AJ Schmidt (https://github.com/ajschmidt8)
  - Dante Gama Dessavre (https://github.com/dantegd)

URL: https://github.com/rapidsai/cuml/pull/4396

achirkin committed:
Add a warning to prefer LinearSVM over SVM(kernel='linear') (#4382)

Suggest using LinearSVM when the user chooses to use the linear kernel in SVM. The reason is that LinearSVM uses a specialized faster solver.

Closes #1664
Also partially addresses #2857

Authors:
  - Artem M. Chirkin (https://github.com/achirkin)

Approvers:
  - Tamas Bela Feher (https://github.com/tfeher)
  - Dante Gama Dessavre (https://github.com/dantegd)

URL: https://github.com/rapidsai/cuml/pull/4382

cjnolet committed:
Hiding cusparse deprecation warnings (#4373)



Authors:
  - Corey J. Nolet (https://github.com/cjnolet)

Approvers:
  - Dante Gama Dessavre (https://github.com/dantegd)

URL: https://github.com/rapidsai/cuml/pull/4373

Commit:
fix minor ASAN issues in UMAPAlgo::Optimize::find_params_ab() (#4405)

There were actuall 2 minor issues that prevented `UMAPAlgo::Optimize::find_params_ab()` from being ASAN-clean at the moment:

- One is the mem leaks, of course
- Another one is the `malloc()`-`delete` mismatch -- only memory allocated using `new` or equivalent should be freed with operator `delete` or `delete[]`

Another issue that was also addressed here: exception safety (i.e., by using `make_unique` from C++-14)

Signed-off-by: Yitao Li <yitao@rstudio.com>

Authors:
  - Yitao Li (https://github.com/yitao-li)

Approvers:
  - Zach Bjornson (https://github.com/zbjornson)
  - Corey J. Nolet (https://github.com/cjnolet)

URL: https://github.com/rapidsai/cuml/pull/4405

zbjornson committed:
Simplify perplexity normalization in t-SNE (#4425)

P_sum is equal to n. See #2622 where I made this change once before. #4208 changed it back while consolidating code.

Authors:
  - Zach Bjornson (https://github.com/zbjornson)

Approvers:
  - Corey J. Nolet (https://github.com/cjnolet)

URL: https://github.com/rapidsai/cuml/pull/4425

venkywonka committed:
RF: code re-organization to enhance build parallelism (#4299)

This PR separates the Decision tree kernels into separate Translation Units (TU) and explicitly instantiates templates. 
This is helpful in 2 ways:
1. refactoring top-level RF/DT code now would not require recompilation of the kernels 
2. Since they are separated into different TUs and linked, they can leverage build parallelism (4x improvement in rebuild times after touching kernel definitions)

Rebuilding by running `time ./build.sh libcuml -v -n PARALLEL_LEVEL=20` after touching RF kernels comparison:
(Note: using `--ccache` doesn't matter here, assuming after touching RF kernels the state of the code-base is completely new and not part of ccache's hashed index)
<details><summary>This PR</summary>
  
  ```
  real    0m20.054s             
  user    2m28.436s                                               
  sys     0m14.241s 
  ```
</details>
<details><summary>branch-21.12</summary>
  
  ```
real    1m21.197s                                                                                                                    
user    2m5.751s                                                                                                                     
sys     0m6.050s
  ```
</details>

Some other changes include renaming and reorganizing files, pruning headers and cleaning up some code

Things to do: 
- [x] split DT Kernels
- [x] benchmark for regressions

Authors:
  - Venkat (https://github.com/venkywonka)

Approvers:
  - Rory Mitchell (https://github.com/RAMitchell)
  - Dante Gama Dessavre (https://github.com/dantegd)

URL: https://github.com/rapidsai/cuml/pull/4299

viclafargue committed:
Remove comment numerical warning (#4408)

Answers #4203
Just set in stone the warning filter for "Numerical issues".

Authors:
  - Victor Lafargue (https://github.com/viclafargue)

Approvers:
  - Dante Gama Dessavre (https://github.com/dantegd)

URL: https://github.com/rapidsai/cuml/pull/4408

hcho3 committed:
Fix docstring for npermutations in PermutationExplainer (#4402)

Closes #4047

Authors:
  - Philip Hyunsu Cho (https://github.com/hcho3)

Approvers:
  - Dante Gama Dessavre (https://github.com/dantegd)

URL: https://github.com/rapidsai/cuml/pull/4402

cjnolet committed:
Using sparse public API functions from RAFT (#4389)



Authors:
  - Corey J. Nolet (https://github.com/cjnolet)

Approvers:
  - Divye Gala (https://github.com/divyegala)
  - Dante Gama Dessavre (https://github.com/dantegd)

URL: https://github.com/rapidsai/cuml/pull/4389

dantegd committed:
Remove direct Scikit-learn imports (#4431)

* FIX Remove hard sklearn imports

* FIX Missing whitespace

* FIX minor error

* FIX PEP8 fixes

GPUtester committed:
Merge pull request #4434 from rapidsai/branch-21.12

[gpuCI] Forward-merge branch-21.12 to branch-22.02 [skip gpuci]

divyegala committed:
Integrating RAFT handle updates (#4313)



Authors:
  - Divye Gala (https://github.com/divyegala)
  - Corey J. Nolet (https://github.com/cjnolet)

Approvers:
  - Corey J. Nolet (https://github.com/cjnolet)

URL: https://github.com/rapidsai/cuml/pull/4313

jjacobelli committed:
Update ucx-py version on release using rvc (#4411)

Update `ucx-py` version on release using `rvc`

Authors:
  - Jordan Jacobelli (https://github.com/Ethyling)

Approvers:
  - AJ Schmidt (https://github.com/ajschmidt8)

URL: https://github.com/rapidsai/cuml/pull/4411

bdice committed:
Update CUDA 11.5 conda environment to use 22.02 pinnings. (#4450)

This PR updates the pinnings of the conda environment for CUDA 11.5 to use 22.02 packages. This resolves conflicts between a5e7cfb5c32f6e7eb7e8d64628f686c3b973789e and #4364.

Authors:
  - Bradley Dice (https://github.com/bdice)

Approvers:
  - Dante Gama Dessavre (https://github.com/dantegd)
  - AJ Schmidt (https://github.com/ajschmidt8)

URL: https://github.com/rapidsai/cuml/pull/4450

cjnolet committed:
Use RAFT template instantations for distances (#4302)



Authors:
  - Corey J. Nolet (https://github.com/cjnolet)
  - Dante Gama Dessavre (https://github.com/dantegd)

Approvers:
  - Divye Gala (https://github.com/divyegala)
  - Dante Gama Dessavre (https://github.com/dantegd)
  - Jordan Jacobelli (https://github.com/Ethyling)

URL: https://github.com/rapidsai/cuml/pull/4302

levsnv committed:
Merge branch 'branch-22.02' of github.com:rapidsai/cuml into unify-tests

jjacobelli committed:
Add missing imports tests (#4452)

Add missing imports tests

Authors:
  - Jordan Jacobelli (https://github.com/Ethyling)

Approvers:
  - AJ Schmidt (https://github.com/ajschmidt8)

URL: https://github.com/rapidsai/cuml/pull/4452

levsnv committed:
Merge branch 'unify-dense-sparse-import' into unify-tests

levsnv committed:
moved node_traits from common.cuh to internal.cuh to use in fil_tests.cu

levsnv committed:
types

levsnv committed:
ref -> val

achirkin committed:
Move NVTX range helpers to raft (#4445)

Move NVTX range helpers to raft  and extend them a little bit.
Corresponding raft PR: https://github.com/rapidsai/raft/pull/416 .

Authors:
  - Artem M. Chirkin (https://github.com/achirkin)

Approvers:
  - Corey J. Nolet (https://github.com/cjnolet)

URL: https://github.com/rapidsai/cuml/pull/4445

levsnv committed:
Unify dense and sparse import in FIL (#4328)



Authors:
  - Levs Dolgovs (https://github.com/levsnv)

Approvers:
  - Andy Adinets (https://github.com/canonizer)
  - William Hicks (https://github.com/wphicks)

URL: https://github.com/rapidsai/cuml/pull/4328

levsnv committed:
Merge branch 'branch-22.02' into unify-tests

levsnv committed:
fix conflict resolution
